src/ast/passes/named_param.cpp

```diff
@@ -0,0 +1,128 @@
+  using Visitor<NamedParamPass>::visit;
+  void visit(Expression &expr);
+
+  std::unordered_map<std::string, std::string> used_args;
+
+private:
+  ASTContext &ast_;
+  BPFtrace &bpftrace_;
+};
+
+void NamedParamPass::visit(Expression &expr)
+{
+  auto *call = expr.as<Call>();
+  if (!call || call->func != "getopt") {
+    Visitor<NamedParamPass>::visit(expr);
+    return;
+  }
+
+  auto *arg_name = call->vargs.at(0).as<String>();
+  if (!arg_name) {
+    call->vargs.at(0).node().addError()
+        << "First argument to 'getopt' must be a string literal.";
+    return;
+  }
+
+  bool is_arg_error = false;
+  if (call->vargs.size() == 2) {
+    if (!call->vargs.at(1).as<Integer>() &&
+        !call->vargs.at(1).as<NegativeInteger>() &&
+        !call->vargs.at(1).as<String>() &&
+        !call->vargs.at(1).as<Identifier>()) {
+      is_arg_error = true;
+    } else if (auto *ident = call->vargs.at(1).as<Identifier>()) {
+      if (!util::is_str_bool_truthy(ident->ident) &&
+          !util::is_str_bool_falsy(ident->ident)) {
+        is_arg_error = true;
+      }
+    }
+    if (is_arg_error) {
+      call->vargs.at(1).node().addError()
+          << "Second argument to 'getopt' must be a string literal, integer "
+             "literal, or a boolean-ish identifier (e.g. true, false, yes, "
+             "no).";
+      return;
+    }
+  }
+
+  std::string expected_default_val;
+  if (call->vargs.size() == 1) {
+    // boolean
+    expr.value = ast_.make_node<NamedParameter>(
+        arg_name->value, CreateInt64(), "0", true, Location(call->loc));
+    expected_default_val = "false";
+  } else if (auto *default_value = call->vargs.at(1).as<String>()) {
+    // string
+    expr.value = ast_.make_node<NamedParameter>(
+        arg_name->value,
+        CreateString(bpftrace_.config_->max_strlen),
+        default_value->value,
+        false,
+        Location(call->loc));
+    expected_default_val = default_value->value;
+  } else if (auto *default_value = call->vargs.at(1).as<Integer>()) {
+    // integer
+    expected_default_val = std::to_string(default_value->value);
+    expr.value = ast_.make_node<NamedParameter>(arg_name->value,
+                                                CreateInt64(),
+                                                expected_default_val,
+                                                false,
+                                                Location(call->loc));
+  } else if (auto *default_value = call->vargs.at(1).as<NegativeInteger>()) {
+    // integer
+    expected_default_val = std::to_string(default_value->value);
+    expr.value = ast_.make_node<NamedParameter>(arg_name->value,
+                                                CreateInt64(),
+                                                expected_default_val,
+                                                false,
+                                                Location(call->loc));
+  } else if (auto *default_value = call->vargs.at(1).as<Identifier>()) {
```

Comment:
I'm not a huge fan of the magic identifier cases (there are only a few, and this is adding a new one). Maybe you could file an issue to make true/false actual values that are passed as a boolean type, and we could at least leave a FIXME here? I wouldn't advertise support for `yes`/`no` as part of the API.